src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -781,6 +780,13 @@ _test_end_sessions_many (bool pooled)
     */
    for (i = 0; i < sizeof sessions / sizeof (mongoc_client_session_t *); i++) {
       sessions[i] = mongoc_client_start_session (client, NULL, &error);
+      mongoc_client_command_with_opts (client,
+                                       "admin",
+                                       tmp_bson ("{'ping': 1}"),
```

Comment:
This is more of an internal detail. But instead of using `tmp_bson` in the loop, I suggest creating a `bson_t` another way.

`tmp_bson` is a test convenience. Each time it is called, a `bson_t` is allocated on the heap and added to a global list. Currently, this does not get cleared until process exit. I'm not sure if there'd be any observable performance hit for the test runner, but creating 10,000 seems unnecessary.

If my suggestion below works out (sending ping instead of modifying `last_used_usec`), I suggest adding a helper to send `ping` and using that in both places. This could construct the ping command without `tmp_bson`:

```
static void send_ping (mongoc_client_t *client, mongoc_client_session_t *client_session) {
   bson_t ping_cmd = BSON_INITIALIZER;
   
   BCON_APPEND (&cmd, "ping", BCON_INT32 (1));
   /* ... */
}
```